example/cajita_tutorial/10_spline/spline_example.cpp

```diff
@@ -0,0 +1,229 @@
+        global_low_corner[2] + cell_size * global_num_cell[2] };
+
+    // Create the global mesh
+    auto global_mesh = Cajita::createUniformGlobalMesh(
+        global_low_corner, global_high_corner, cell_size );
+
+    // Set up a uniform partitioner
+    Cajita::DimBlockPartitioner<3> partitioner;
+
+    // Create the global grid.
+    auto global_grid = Cajita::createGlobalGrid(
+        MPI_COMM_WORLD, global_mesh, { false, false, false }, partitioner );
+
+    // Get the local grid.
+    int halo_cell_width = 1;
+    auto local_grid = Cajita::createLocalGrid( global_grid, halo_cell_width );
+
+    // Get the local mesh for this rank.
+    auto local_mesh = Cajita::createLocalMesh<ExecutionSpace>( *local_grid );
+
+    // Create a particle position vector.
+    double xp[3] = { 0.5, 2, 1.5 };
+
+    /********************************************************************/
+    /*
+      There are two ways to use Cajita splines in practice.
+
+      In the first, one constructs an object of type:
+        SplineData<Scalar, Order, NumSpaceDim, EntityType, Tags...> where
+            Scalar is the underlying scalar type
+            Order = spline order
+            NumSpaceDim = number of spatial dimensions
+            EntityType = geometric entity type over which the spline is defined
+            Tags = variadic list of desired SplineDataMember tags.
+
+      The available SplineDataMembers and their associated member names are
+        SplinePhysicalCellSize = dx[d]
+        SplineLogicalPosition = x[d]
+        SplinePhysicalDistance = d[d]
+        SplineWeightValues = w[d][num_knots]
+        SplineWeightPhysicalGradients g[d][num_knots]
+
+      This design allows for customizing the relavant spline data for a
+      particular application. If all Tags are omitted, all SplineDataMembers are
+      included.
+    */
+    using value_type = double;
+    constexpr int order = 2;     // quadratic B-spline
+    constexpr int num_space = 3; // 3-dimensional
+
+    // This spline data type will be defined on grid nodes in this example.
+    using SD = Cajita::SplineData<value_type, order, num_space, Cajita::Node>;
+
+    SD sd_n; // Default construction
+    std::cout << "Spline order: " << SD::order << " (quadratic)" << std::endl;
+
+    // Evaluate the spline data given a point position and the local mesh
+    Cajita::evaluateSpline( local_mesh, xp, sd_n );
+
+    // The spline data by default contains all available SplineDataMembers,
+    // whose values were all updated from the previous call to evaluateSpline()
+    std::cout << "Spline weights:" << std::endl;
+    for ( int d = 0; d < num_space; ++d )
+    {
+        std::cout << "  Dim " << d << ": ";
+        for ( int k = 0; k < SD::num_knot; ++k )
+        {
+            std::cout << sd_n.w[d][k] << " ";
+        }
+        std::cout << std::endl;
+    }
+    std::cout << std::endl;
+
+    /*
+      We can also use the Spline<Order> interface directly to evaluate
```

Comment:
Here I would point out that this is useful if you are writing your own particle-grid operators whereas using the existing particle-grid operators in Cajita would not require use of these interfaces beyond simply constructing the spline data.

Reply:
Done.